src/components/Steps/index.js

```diff
@@ -14,108 +15,114 @@ class Steps extends React.Component {
+                ...step,
+                ref: React.createRef(),
+                active: false,
+                errors: [],
+            })),
+            validated: false,
         };
+
+        this.activateStep(0);
     }
 
-    scrollTo =
-        ref => (ref.current
-            ? scrollToComponent(ref.current, {
-                offset: -50,
-                align: 'top',
-                duration: 1000,
-            })
-            : {});
+    componentDidUpdate() {
+        const lastActivatedStepIndex = this.state.steps.filter(step => step.active).length - 1;
+        this.scrollTo(this.state.steps[lastActivatedStepIndex]);
+    }
 
-    isComponentToShow = (name) => {
-        switch (name) {
-        case 'Choose your job/school address':
-            return this.props.city.name;
-        case 'Choose additional criteria':
-            return !!this.props.city.localization;
-        case 'Map':
-            return !!this.props.criteria;
-        default:
-            return true;
-        }
-    };
+    setErrors = (index, errors) => {
+        const { steps } = this.state;
+        steps[index].errors = errors;
+        this.setState({ steps });
+    }
 
-    displayMap = () => {
-        this.setState({
-            map: true,
-        });
-    };
+    activateStep = (index) => {
+        const { steps } = this.state;
+        steps[index].active = true;
+        this.setState({ steps });
+    }
 
-    render() {
-        const steps = [
-            {
-                name: 'Choose city',
-                component: SelectCityStep,
-            },
-            {
-                name: 'Choose your job/school address',
-                component: SearchStep,
-            },
-            {
-                name: 'Choose additional criteria',
-                component: CriteriaStep,
-            },
-        ];
+    scrollTo = ({ ref }) => (ref.current
+        ? scrollToComponent(ref.current, {
+            offset: -50,
+            align: 'top',
+            duration: 1000,
+        })
+        : {});
 
-        if (this.state.map) {
-            steps.push({
-                name: 'Map',
-                component: MapStep,
-            });
+    validate = () => {
+        this.setState({ validated: true });
+        const firstErrorStep = this.state.steps
+            .filter(step => !step.component.validate(this.props))[0];
+        if (firstErrorStep) {
+            this.scrollTo(firstErrorStep);
+            return false;
         }
+        return true;
+    };
 
+    render() {
         return (
-
             <div className="step-container">
                 <ReactTooltip />
-                {steps.filter(({ name }, index) => this.isComponentToShow(name)).map(({ name, component: StepComponent }, index) => {
-                    const prevStep = this.state.steps[index - 1];
-                    const nextStep = this.state.steps[index + 1];
+                {this.state.steps.filter(step => step.active)
```

Comment:
steps to const

Reply:
I need steps in state because I am setting state to activate and validate them.